hamilton/function_modifiers/macros.py

```diff
@@ -1055,3 +1174,345 @@ def chain_transforms(
+            )
+
+        def foo(a:int)->Dict[str,int]:
+            return {"field_1":1, "field_2":2, "field_3":3}
+    """
+
+    def __init__(
+        self,
+        *target_functions: Union[RemoteApplicableBuilder, Callable],
+        collapse: bool = False,
+        _chain: bool = False,
+        **mutating_function_kwargs: Union[SingleDependency, Any],
+    ):
+        """Instantiates a `\@mutate` decorator.
+
+        We assume the first argument of the decorated function to be the output of the function we are targeting.
+
+        :param target_functions: functions we wish to mutate the output of
+        :param collapse: Whether to collapse this into a single node. This is not currently supported.
+        :param _chain: Whether to chain the first parameter. This is the only mode that is supported. Furthermore, this is not externally exposed. @flow will make use of this.
+        :param \*\*mutating_function_kwargs: other kwargs that the decorated function has
+        """
+        self.collapse = collapse
+        self.chain = _chain
+        # keeping it here once it gets implemented maybe nice to have options
+        if self.collapse:
+            raise NotImplementedError(
+                "Collapsing functions as one node is not yet implemented for mutate(). Please reach out if you want this feature."
+            )
+        if self.chain:
+            raise NotImplementedError("@flow() is not yet supported -- this is ")
+
+        self.remote_applicables = tuple(
+            [apply_to(fn) if isinstance(fn, Callable) else fn for fn in target_functions]
+        )
+        self.mutating_function_kwargs = mutating_function_kwargs
+        self.restrict_to_single_module = True
+
+    def validate_same_module(self, mutating_fn: Callable):
+        """Validates target functions are in the same module as the mutator function.
+
+        :param mutating_fn: Function to validate against
+        :return: Nothing, raises exception if not valid.
+        """
+        local_module = mutating_fn.__module__
+        for remote_applicable in self.remote_applicables:
+            if remote_applicable.target_fn.__module__ != local_module:
+                raise NotSameModuleError(fn=mutating_fn, target_fn=remote_applicable.target_fn)
+
+    def _create_step(
+        self, mutating_fn: Callable, remote_applicable_builder: RemoteApplicableBuilder
+    ):
+        """Resolves the user specific inputs and creates the applicable.
+        This de-facto allows the user to use a 'remote step()' function.
+        """
+        if not remote_applicable_builder.mutating_fn_kwargs:
+            remote_applicable_builder.mutating_fn_kwargs = self.mutating_function_kwargs
+
+        applicable = step(mutating_fn, **remote_applicable_builder.mutating_fn_kwargs)
+
+        if remote_applicable_builder.when_key_value_pairs is not None:
+            applicable = applicable.when(**remote_applicable_builder.when_key_value_pairs)
```

Comment:
Yep, so I think if we make applicable allowed to have a `None` function it might just solve this? Otherwise it's the same, right?